compiler/src/iree/compiler/Dialect/HAL/Target/LLVMCPU/LLVMCPUTarget.cpp

```diff
@@ -444,17 +445,20 @@ class LLVMCPUTargetBackend final : public TargetBackend {
              << targetTriple.str() << "'";
     }
 
-    // Link musl last and pull in all of it - this is sad but LLVM will take IR
-    // intrinsics and generate calls out to libc during code generation and we
-    // have no control over that - if we don't provide the symbols here then
-    // linking with ld will fail.
-    if (failed(linkBitcodeModule(
-            variantOp.getLoc(), moduleLinker, llvm::Linker::OverrideFromSrc,
-            *targetMachine, "libmusl",
-            loadMuslBitcode(targetMachine.get(), context)))) {
-      return mlir::emitError(variantOp.getLoc())
-             << "failed linking in builtin library for target triple '"
-             << targetTriple.str() << "'";
+    if (options_.linkEmbedded) {
+      // Link musl last and pull in all of it - this is sad but LLVM will take
+      // IR
+      // intrinsics and generate calls out to libc during code generation and we
+      // have no control over that - if we don't provide the symbols here then
+      // linking with ld will fail.
+      if (failed(linkBitcodeModule(
```

Comment:
Do we actually want to condition linking bitcode on this? I was under the impression (back from https://github.com/openxla/iree/pull/7721 and others) that we wanted to always link musl / the builtin library, to avoid any system dependencies (so artifacts are more portable and don't suffer from performance penalties from calling out to the system).

Reply:
yes, this isnt changing the default. `options_.linkEmbedded` is by default `true`. When set to `false` we avoid linking musl generated files. This is for cases where you generate a library from IREE and link it with `libm`. In those cases we dont need the implementation from musl.

Reply:
Can we safely assume that system linking has access to libm? (LLVM appears to assume libm's existence in it's lowering, but I thought we wanted to more portable than that, even with the system linking mode)

Reply:
Well, I don't know how important that support point is. We can say you don't need libm and we use musl to get the implementation. Or we say we won't use musl, but you need a libm. 
First though, checking if this fixes the problem. It is really a Riscv backend issue that we are trying to work around .

Reply:
Okay, I don't fully understand all the implications for each platform / configuration, so this might be fine... could check with Ben when he's around.

Reply:
Agreed. I will wait for @benvanik to see if this is OK.